src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportOrchestratorTask.cs

```diff
@@ -122,30 +122,48 @@ public async Task<TaskResultData> ExecuteAsync()
                     await _importOrchestratorTaskDataStoreOperation.PreprocessAsync(cancellationToken);
 
                     _orchestratorTaskContext.Progress = ImportOrchestratorTaskProgress.PreprocessCompleted;
+                    _orchestratorTaskContext.CurrentSequenceId = _sequenceIdGenerator.GetCurrentSequenceId();
                     await UpdateProgressAsync(_orchestratorTaskContext, cancellationToken);
 
                     _logger.LogInformation("Preprocess Completed");
                 }
 
-                if (_orchestratorTaskContext.Progress == ImportOrchestratorTaskProgress.PreprocessCompleted)
+                if (_orchestratorInputData.StoreProgressInSubTask)
```

Comment:
Is this used to distinguish new version from old version? What will happen if we upgrade StoreProgressInSubTask from `false` to `true`.

Reply:
this flag persists in the task payload, old task would not have this task, and created task would not change later.